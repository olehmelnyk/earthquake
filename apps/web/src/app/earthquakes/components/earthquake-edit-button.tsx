import { Edit } from 'lucide-react';
import { Button } from '@earthquake/ui';

export interface EarthquakeEditButtonProps {
  readonly id: string;
  readonly onEdit: (id: string) => void;
}

export const EarthquakeEditButton = ({ id, onEdit }: EarthquakeEditButtonProps) => (
  <Button
    variant="ghost"
    size="icon"
    onClick={() => onEdit(id)}
    className="h-8 w-8"
  >
    <Edit className="h-4 w-4" />
    <span className="sr-only">Edit</span>
  </Button>
);