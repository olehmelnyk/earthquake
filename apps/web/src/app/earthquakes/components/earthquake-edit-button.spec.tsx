import { render, screen } from '@testing-library/react';
import { EarthquakeEditButton } from './earthquake-edit-button';
import { describe, it, expect, vi } from 'vitest';
import { userEvent } from '@testing-library/user-event';

// Mock Lucide icons
vi.mock('lucide-react', () => ({
  Edit: () => <span data-testid="edit-icon">Edit</span>
}));

// Mock UI components
vi.mock('@earthquake/ui', () => ({
  Button: ({
    children,
    onClick,
    variant,
    size,
    className
  }: {
    children: React.ReactNode;
    onClick?: () => void;
    variant?: string;
    size?: string;
    className?: string;
  }) => (
    <button
      onClick={onClick}
      data-variant={variant}
      data-size={size}
      className={className}
      data-testid="edit-button"
    >
      {children}
    </button>
  )
}));

describe('EarthquakeEditButton', () => {
  const mockId = '123';
  const mockOnEdit = vi.fn();

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should render the edit button with icon', () => {
    render(<EarthquakeEditButton id={mockId} onEdit={mockOnEdit} />);

    const button = screen.getByTestId('edit-button');
    expect(button).toBeInTheDocument();

    const icon = screen.getByTestId('edit-icon');
    expect(icon).toBeInTheDocument();

    // Use a more specific selector for the sr-only span
    const srOnlyText = screen.getByText('Edit', { selector: 'span.sr-only' });
    expect(srOnlyText).toBeInTheDocument();
  });

  it('should call onEdit with id when button is clicked', async () => {
    const user = userEvent.setup();

    render(<EarthquakeEditButton id={mockId} onEdit={mockOnEdit} />);

    const button = screen.getByTestId('edit-button');
    await user.click(button);

    expect(mockOnEdit).toHaveBeenCalledTimes(1);
    expect(mockOnEdit).toHaveBeenCalledWith(mockId);
  });

  it('should have correct variant and size props', () => {
    render(<EarthquakeEditButton id={mockId} onEdit={mockOnEdit} />);

    const button = screen.getByTestId('edit-button');
    expect(button).toHaveAttribute('data-variant', 'ghost');
    expect(button).toHaveAttribute('data-size', 'icon');
    // Check className property instead of attribute
    expect(button.className).toBe('h-8 w-8');
  });

  it('should match snapshot', () => {
    const { container } = render(<EarthquakeEditButton id={mockId} onEdit={mockOnEdit} />);
    expect(container).toMatchSnapshot();
  });
});