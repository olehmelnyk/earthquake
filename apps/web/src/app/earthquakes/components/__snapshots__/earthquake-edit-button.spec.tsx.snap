// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EarthquakeEditButton > should match snapshot 1`] = `
<div>
  <button
    class="h-8 w-8"
    data-size="icon"
    data-testid="edit-button"
    data-variant="ghost"
  >
    <span
      data-testid="edit-icon"
    >
      Edit
    </span>
    <span
      class="sr-only"
    >
      Edit
    </span>
  </button>
</div>
`;
